test/datasets/test_udpos.py

```diff
@@ -0,0 +1,87 @@
+import os
+import random
+import string
+import zipfile
+from collections import defaultdict
+from unittest.mock import patch
+
+from parameterized import parameterized
+from torchtext.datasets.udpos import UDPOS
+
+from ..common.case_utils import TempDirMixin, zip_equal
+from ..common.torchtext_test_case import TorchtextTestCase
+
+
+def _get_mock_dataset(root_dir):
+    """
+    root_dir: directory to the mocked dataset
+    """
+    base_dir = os.path.join(root_dir, "UDPOS")
+    temp_dataset_dir = os.path.join(base_dir, "temp_dataset_dir")
+    os.makedirs(temp_dataset_dir, exist_ok=True)
+
+    seed = 1
+    mocked_data = defaultdict(list)
+    for file_name in ["train.txt", "dev.txt", "test.txt"]:
+        txt_file = os.path.join(temp_dataset_dir, file_name)
+        mocked_lines = mocked_data[os.path.splitext(file_name)[0]]
+        with open(txt_file, "w") as f:
+            for i in range(5):
+                label = seed % 2
+                rand_strings = [random.choice(string.ascii_letters) for i in range(seed)]
+                rand_label_1 = [random.choice(string.ascii_letters) for i in range(seed)]
+                rand_label_2 = [random.choice(string.ascii_letters) for i in range(seed)]
+                # one token per line (each sample ends with an extra \n)
+                for rand_string, label_1, label_2 in zip(rand_strings, rand_label_1, rand_label_2):
+                    f.write(f"{rand_string}\t{label_1}\t{label_2}\n")
+                f.write("\n")
```

Comment:
Maybe worth adding more examples than a single one to make sure the IOB reading for >1 example is tested too.

Reply:
Having a look at the [udpos data](https://bitbucket.org/sivareddyg/public/downloads/en-ud-v2.zip), it is tokenized per line, so in my case each token has one character and is on one line. I assume this is what you meant by example, so I changed to randomly sample 1 to 10 characters instead for each token.

Reply:
By examples I mean "sentences" which are newline delimited:

```
a this this
b is is
c example example
d one one

e this this
f is is
g example example
h two two
```

Reply:
Ah, it looks OK. I somehow missed that that these were lists.